src/System.IO.FileSystem/src/System/IO/Win32FileSystemObject.cs

```diff
@@ -64,7 +64,7 @@ public DateTimeOffset CreationTime
                 {
                     EnsureDataInitialized();
                     long dt = ((long)(_data.ftCreationTimeHigh) << 32) | ((long)_data.ftCreationTimeLow);
-                    return DateTimeOffset.FromFileTime(dt);
+                    return DateTimeOffset.FromFileTime(dt).ToLocalTime();
```

Comment:
Windows DateTimeOffset.FromFileTime is working the same way that is on Desktop: FromFileTime returns a DateTimeOffset with a properly set time. The issue on Windows is only in File.cs where the DateTimeOffset is being converted into a DateTime using the .DateTime property (DateTimeKind.Unspecified) instead of the .LocalTime property (DateTimeKind.Local).